x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/timing_footers/timing_footers_context.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { createContext, FunctionComponent, ReactNode, useContext, useMemo } from 'react';
+import { UseField, useFormData } from '../../../../../shared_imports';
+import { FormInternal } from '../../types';
+import { calculateRelativeTimingMs, normalizeTimingsToHumanReadable } from '../../lib';
+import { TimingFooter } from './timing_footer';
+
+export interface TimingFooters {
+  hot: ReactNode;
+  warm: ReactNode;
+  cold: ReactNode;
+  setDeletePhaseEnabled: (value: boolean) => void;
+}
+
+const TimingFootersContext = createContext<TimingFooters>(null as any);
+
+export const TimingFootersProvider: FunctionComponent = ({ children }) => {
+  const [formData] = useFormData<FormInternal>();
+
+  const phaseTimings = useMemo(() => {
+    return calculateRelativeTimingMs(formData);
+  }, [formData]);
+
+  const humanReadableTimings = useMemo(() => normalizeTimingsToHumanReadable(phaseTimings), [
+    phaseTimings,
+  ]);
+
+  const createPhaseTimingFooter = (
+    phase: 'hot' | 'warm' | 'cold',
+    setValue: (value: boolean) => void
+  ) => (
+    <TimingFooter
+      timingInMs={phaseTimings.phases[phase]}
+      timingLabel={humanReadableTimings[phase]}
+      setValue={setValue}
+    />
+  );
+
+  return (
+    <UseField path={'_meta.delete.enabled'}>
```

Comment:
Nit; for here and elsewhere; we usually write static string props using this pattern:

```ts
<UseField path="_meta.delete.enabled">
```